drivers/srf04/srf04.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup drivers_srf04
+ * @{
+ *
+ * @file
+ * @brief       driver for srf04 ultra sonic range finder
+ *
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "srf04.h"
+#include "srf04_params.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static void _cb(void *arg)
+{
+    uint32_t t = xtimer_now_usec();
+
+    srf04_t* dev = (srf04_t*)arg;
+    if (dev->distance > SRF04_ERR_MEASURING) {
+        dev->distance = SRF04_ERR_MEASURING;
+        dev->time = t;
+    } else {
+        gpio_irq_disable(dev->p.echo);
+        dev->distance = (t - dev->time);
+    }
+}
+
+int srf04_init(srf04_t* dev)
+{
+    dev->p = srf04_params[0];
```

Comment:
This limits the driver to a single hardware sensor. I suggest adding an argument to the init function, passing a pointer to a params struct. See other device drivers in the tree, for example https://github.com/RIOT-OS/RIOT/blob/master/drivers/io1_xplained/io1_xplained.c#L37-L39